camply/notifications/discord.py

```diff
@@ -0,0 +1,113 @@
+    """
+    Push Notifications via Discord
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.session.headers.update({"Content-Type": "application/json"})
+        if any([DiscordConfig.DISCORD_WEBHOOK is None, DiscordConfig.DISCORD_WEBHOOK == ""]):
+            warning_message = (
+                "Discord is not configured properly. To send Discord messages "
+                "make sure to run `camply configure` or set the "
+                "proper environment variable: `DISCORD_WEBHOOK`."
+            )
+            logger.error(warning_message)
+            raise EnvironmentError(warning_message)
+
+    def send_message(self, message: str, **kwargs) -> requests.Response:
+        """
+        Send a message via Discord - if environment variables are configured.
+
+        Parameters
+        ----------
+        message: str
+
+        Returns
+        -------
+        requests.Response
+        """
+        message_json = kwargs
+        if message:
+            message_json["content"] = message
+
+        logger.debug(message_json)
+        response = self.session.post(
+            url=DiscordConfig.DISCORD_WEBHOOK,
+            json=message_json,
+        )
+        try:
+            response.raise_for_status()
+        except requests.HTTPError as he:
+            logger.warning(
+                "Notifications weren't able to be sent to Discord. "
+                "Your configuration might be incorrect."
+            )
+            raise ConnectionError(response.text) from he
+        return response
+
+    def block_for_campsite(self, campsite: AvailableCampsite):
+        message_title, formatted_dict = self.format_standard_campsites(
+            campsite=campsite,
+        )
+
+        # Remove items that will be templated as part of the embed.
+        del formatted_dict["Recreation Area"]
+        del formatted_dict["Booking Date"]
+        del formatted_dict["Booking End Date"]
+        del formatted_dict["Facility Name"]
+        del formatted_dict["Booking Link"]
+        del formatted_dict["Campsite Site Name"]
+        del formatted_dict["Campsite Loop Name"]
+        del formatted_dict["Recreation Area Id"]
+        del formatted_dict["Facility Id"]
+        del formatted_dict["Campsite Id"]
+
+        return {
+            "author": {
+                "name": f"🏕 {campsite.recreation_area}"
+            },
+            "title": f"{campsite.facility_name} {campsite.campsite_loop_name} #{campsite.campsite_site_name}",
+            "description": f"{campsite.booking_date:%Y/%m/%d} to {campsite.booking_end_date:%Y/%m/%d}",
+            "url": campsite.booking_url,
+            "color": 2375436,
+            "fields": [
+                {
+                    "name": key,
+                    "value": str(value)
+                } for key, value in formatted_dict.items()
+            ],
+            "footer": {
+                "text": "camply, the campsite finder ⛺️"
+            }
+        }
```

Comment:
A couple notes on this - I think there are a good number of fields that we want to keep from our `formatted_dict` here - although it can be quite verbose when looking at the notifications the extra context can be useful. 

- This `title` field here assumes that the campsite has a numeric `campsite_loop_name` - but in a good number of campsites this is a string value. 
- There are provider's who don't allow the passing of campsite ID via the booking URL - for those cases we'll need to keep the campsite ID in the message
- Is it possible to have a notification open in a "collapsed" view that can be expanded? This would be nice to have the default view be simple, but with expanded details.

See here for the `test-notifications` CLI command:

```shell
camply test-notifications --notifications discord
```

<img width="412" alt="image" src="https://github.com/juftin/camply/assets/49741340/53ac275d-bc06-432e-a8a5-7d880115d7d8">

Here's the same notification, that instead uses `message_title` and doesn't remove all the fields - I think some of these can be omitted though like Booking Date / Booking URL which are provided elsewhere 
<img width="348" alt="image" src="https://github.com/juftin/camply/assets/49741340/520e79ab-20d3-4373-82c5-8f38dd1bca8a">